woocommerce/woocommerce-gutenberg-products-block
# Add Inner Block Rendering and Atomic Block Styles

This is I think the last thing to be pulled out of #2399 and takes care of **Inner Block Rendering.**

- Each Atomic Block is registered and saves content - an empty div with classname.
- Each Atomic Block now has styles and unique, non-prefixed classnames so styles are used regardless of layout container
- `render_frontend` handles the mapping between the saved div and the block component to render the layout
- `withFilteredAttributes` translates dataset attributes from the DOM to block attributes by looking up which attributes are defined and mapping/formatting the values. 

I also did a bulk rename of `wc-block-component-` classes to `wc-block-components-` since this was an oversight.

Closes #2399
Partly tackles #2527 but needs extra testing nearer completion.

### How to test the changes in this Pull Request:

1. All Products Block - No regressions, in particular styling, across the main themes (I am checking this some more also since there are lots of combinations)
2. Single Products Block - Check you can add inner blocks and they are rendered on the frontend with correct styles!
3. Ensure existing all products blocks do not invalidate
